core/src/utils/overlays.ts

```diff
@@ -328,33 +330,25 @@ export const getOverlays = (doc: Document, selector?: string): HTMLIonOverlayEle
 };
 
 /**
- * Gets the top-most/last opened
- * overlay that is currently presented.
+ * Returns an overlay element
+ * @param doc The document to find the element within.
+ * @param overlayTag The selector for the overlay, defaults to Ionic overlay components.
+ * @param id The unique identifier for the overlay instance.
+ * @param includeHidden `true` to include hidden overlays. `true` by default.
+ * @returns The overlay element or `undefined` if no overlay element is found.
  */
-const getTopOpenOverlay = (doc: Document): HTMLIonOverlayElement | undefined => {
-  const overlays = getOverlays(doc);
-  for (let i = overlays.length - 1; i >= 0; i--) {
-    const overlay = overlays[i];
-
-    /**
-     * Only consider overlays that
-     * are presented. Presented overlays
-     * will not have the .overlay-hidden
-     * class on the host.
-     */
-    if (!overlay.classList.contains('overlay-hidden')) {
-      return overlay;
-    }
-  }
-
-  return;
-}
-
-export const getOverlay = (doc: Document, overlayTag?: string, id?: string): HTMLIonOverlayElement | undefined => {
+export const getOverlay = (doc: Document, overlayTag?: string, id?: string, includeHidden = true): HTMLIonOverlayElement | undefined => {
   const overlays = getOverlays(doc, overlayTag);
-  return (id === undefined)
-    ? overlays[overlays.length - 1]
-    : overlays.find(o => o.id === id);
+  if (includeHidden) {
+    return (id === undefined)
+      ? overlays[overlays.length - 1]
+      : overlays.find(o => o.id === id);
+  } else {
+    const visibleOverlays = overlays.filter(o => !isOverlayHidden(o));
+    return (id === undefined)
+      ? visibleOverlays[visibleOverlays.length - 1]
+      : visibleOverlays.find(o => o.id === id);
+  }
```

Comment:
This could be simplified a bit by having a `filteredOverlays` reference that either gets filtered or not depending on `includeHidden`, then using that for either case. That way you don't need to duplicate the rest of the logic.